src/BackgroundProcess/Domain/BatchItem.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace Jetty\BackgroundProcessing\BackgroundProcess\Domain;
+
+/**
+ * A single batch item to be referenced in background processes.
+ */
+interface BatchItem
+{
+    /**
+     * The key of the BatchItem.
+     */
+    public function key(): string;
+
+
+    /**
+     * The value of the BatchItem. This data will be processed by a background job queue.
+     *
+     * @return mixed
+     */
+    public function value();
+}
```

Comment:
Remove the interface, move the `WpBatchItem` class up to the domain level, and drop "Wp" from the class's name. The model is doing nothing with WordPress. It is a model (a container) for the values returned by the Repository (more on that later). Rather, this model is defining a `KeyValuePair`, which is a common entity in other languages (and a good choice for a name, too).

Now, on the other hand, the `WpBatchItemRepository` _is_ WordPress-specific, because it is interacting with the WordPress DB helper functions. It _is_ an adapter of the application interface for such. This repository is simply returning the above domain model back to the caller.

Reply:
Done